src/app/(proper_react)/redesign/(authenticated)/user/dashboard/fix/leaked-passwords/LeakedPasswordsLayout.tsx

```diff
@@ -32,59 +33,70 @@ export function LeakedPasswordsLayout(props: LeakedPasswordsLayoutProps) {
   const l10n = useL10n();
 
   const stepMap: Record<LeakedPasswordsTypes, StepLink["id"]> = {
-    password: "LeakedPasswordsPassword",
-    "security-question": "LeakedPasswordsSecurityQuestion",
+    passwords: "LeakedPasswordsPassword",
+    "passwords-done": "LeakedPasswordsPassword",
+    "security-questions": "LeakedPasswordsSecurityQuestion",
+    "security-questions-done": "LeakedPasswordsSecurityQuestion",
+    none: "LeakedPasswordsSecurityQuestion",
   };
 
   const guidedExperienceBreaches = getGuidedExperienceBreaches(
     props.data.subscriberBreaches,
     props.subscriberEmails,
   );
 
+  const nextStep = getNextGuidedStep(props.data, stepMap[props.type]);
   const pageData = getLeakedPasswords({
     dataType: props.type,
     breaches: guidedExperienceBreaches,
     l10n: l10n,
+    nextStep,
   });
 
   // The non-null assertion here should be safe since we already did this check
   // in `./[type]/page.tsx`:
   const { title, illustration, content } = pageData!;
+  const isStepDone =
+    props.type === "passwords-done" || props.type === "security-questions-done";
 
   return (
     <FixView
       subscriberEmails={props.subscriberEmails}
       data={props.data}
-      nextStepHref={getNextGuidedStep(props.data, stepMap[props.type]).href}
+      nextStep={nextStep}
       currentSection="leaked-passwords"
+      hideProgressIndicator={isStepDone}
+      showConfetti={isStepDone}
```

Comment:
Since we're passing in the same value twice here... Wouldn't it make more sense to replace them by one `isStepDone` prop?

(Also, when abstractions start having code paths specific to a single use case, that's usually the point where I take a step back to consider whether that abstraction actually still makes sense. I'm not saying that's not the case here (I think the majority of the code is still shared, and future changes will keep having to apply to all its call sites?), but just wanted to make sure you also gave that some conscious thought :slightly_smiling_face:  )

Reply:
I’ve thought about that as well and already anticipated a comment on that: My reasoning to do it in this way is that the prop names are indicating what the appearance and functionality of the component will be like, which I find a more declarative pattern than deriving those from props. We don’t really have a guideline for that and I’ve been not consistent with that myself, though.